server/index.js

```diff
@@ -127,6 +127,30 @@ export default class Server {
         await this.renderJSON(req, res, pathname)
       },
 
+      '/_next/:buildId/page/:path*': async (req, res, params) => {
+        const paths = params.path || ['']
+        const pathname = `/${paths.join('/')}`
+
+        if (this.dev) {
+          await this.hotReloader.ensurePage(pathname)
+        }
+
+        if (!this.handleBuildId(params.buildId, res)) {
```

Comment:
@bringking That's true.
But detecting 404 between browsers will be an issue.
This is a much better way and we know it's working 100% sure.

And we could improve this later on.